### PR TITLE
Update README.haiku

### PR DESCRIPTION
**Pull request by thibaultduponchelle**
Some more up to date informations about past/present notable difference/known problems.

### CODE REVIEW

File: README.haiku [hunk 1]
```diff
@@ -19,22 +19,38 @@ The build procedure is completely standard:
...
 =item *
 
-Perl cannot be compiled with threading support ATM.
+Haiku changed its filesystem hierarchy between BeOS/first Haiku versions and more recent ones.
+Haiku has a system utility "finddir" which is used to returns these paths, and
+it is now used in haiku hints (starting perl 5.32).
+
+=item *
+
+Haiku file system has a limitation and does not store the access time.
```
Comment (khwilliamson) [inline README.haiku:42]: Change to 'The Haiku file system...'

File: README.haiku [hunk 2]
```diff
@@ -19,22 +19,38 @@ The build procedure is completely standard:
...
+When using "stat", it will return always the local time,
```
Comment (khwilliamson) [inline README.haiku:43]: Change to 'always return', as that is more idiomatic English

File: README.haiku [hunk 3]
```diff
@@ -19,22 +19,38 @@ The build procedure is completely standard:
...
+(like if "stat" was considered as an access).
```
Comment (khwilliamson) [inline README.haiku:44]: Change 'like' to 'as if'

File: README.haiku [hunk 4]
```diff
@@ -19,22 +19,38 @@ The build procedure is completely standard:
...
+It is not the common implementation, but neither not considered false for POSIX.
```
Comment (khwilliamson) [inline README.haiku:45]: I don't understand this sentence.  I think you're saying it is legal in POSIX, but not what one would expect. @Grinnz @jmacdotorg do you have good phrasing for this?

File: README.haiku [hunk 5]
```diff
@@ -19,22 +19,38 @@ The build procedure is completely standard:
...
+The tests relying on atime (e.g. the -X operator "-A")
+have then some unexpected results and need to be skipped (fixed in perl 5.36).
```
Comment (khwilliamson) [inline README.haiku:47]: s/have then/thus have/

File: README.haiku [hunk 6]
```diff
@@ -19,22 +19,38 @@ The build procedure is completely standard:
...
+
+=item *
+
+The ${^CHILD_ERROR_NATIVE} value seems to be the same than exit code.
```
Comment (khwilliamson) [inline README.haiku:51]: 'same than' is not good English.  Perhaps you mean 'same as' ?

File: README.haiku [hunk 7]
```diff
@@ -44,13 +60,14 @@ Haiku yet.
 
 =item *
 
-A subtest of the F<cpan/Sys-Syslog/t/syslog.t> test fails. This is due to Haiku
+A subtest of the F<cpan/Sys-Syslog/t/syslog.t> test was failing. This is due to Haiku
```
Comment (khwilliamson) [inline README.haiku:63]: It has been fixed?

**Review by khwilliamson — CHANGES_REQUESTED** [review]

**Comment by thibaultduponchelle** [comment]
I will close this for now since it needs some updates, has conflicts and I don't want to keep it "stale".

I appreciated the review and I will integrates comments and reopen a clean PR when time comes :grinning:

**Pull request closed without merging**